### ISSUE TITLE
sphinx  index directive fails needing twoColumn

### ISSUE THREAD

**Issue by rogerbinns** [rst2pdf/rst2pdf#1122]
**See later comments as the root cause was figured out**

**Description of problem**

Follow the instructions for sphinx integration, and then try to build pdf.

You will get an error:

    Template 'twoColumn' is not defined

This is as expected because that template has been removed from rst2pdf.  The sample config has:

    pdf_page_template = 'cutePage'

However there is no such template in Sphinx or rst2pdf.  The only templates I can find is for the cover page.  This leaves me stuck because I have no idea what should be in the template, and neither project is providing a default.

🖥 **Versions**

`python -V`
Python 3.10.7

`pip freeze | grep rst2pdf`
rst2pdf==0.99

`pip freeze | grep reportlab`
reportlab==3.6.12

`pip freeze | grep Sphinx`
Sphinx==6.1.3

Which operating system are you using?
Ubuntu 22.10

**Additional information**

It would be great if rst2pdf worked out of the box with Sphinx.

**Comment by lornajane** [comment]
Thanks @rogerbinns , this is a really good bug report. I didn't realise that Sphinx assumed the twocolumn style would be present. Some Sphinx things do work perfectly out of the box, but not this clearly, and we don't have a test that covers this case. Can you show a simple replication case and how you would invoke sphinx to create the PDF? I would be interested to look more closely at this one, and it might be that we need to reinstate that old template or something.

**Comment by rogerbinns** [comment]
That was a wild ride.  A completely fresh project made using `sphinx-quickstart` and following [rst2pdf instructions](https://github.com/rst2pdf/rst2pdf/blob/main/doc/manual.rst#sphinx) works fine.  I tracked down what causes the "twoColumn" error to appear: if you have this in the text (any number causes the issue).

     :pep:`249`

I did try removing all the `:pep:` but the error still happens, so there are probably other roles that trigger the problem.

Attached is the minimal reproducer.

[demo.tar.gz](https://github.com/rst2pdf/rst2pdf/files/10892631/demo.tar.gz)

**Comment by rogerbinns** [comment]
I did some more bisection to find why twoColumn coming up.  In addition to :pep: , having [index generating markup](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#index-generating-markup) also causes the problem.  rst2pdf pdfbuilder.py seems a likely culprit

**Comment by rogerbinns** [comment]
Found the final cause and it is indeed [pdfbuilder..py](https://github.com/rst2pdf/rst2pdf/blob/main/rst2pdf/pdfbuilder.py) where it generates indices for Sphinx.  Regular rst doesn't have any index generating markup, while it is very useful in Sphinx.  `twoColumn` was removed as part of https://github.com/rst2pdf/rst2pdf/pull/988

Two locations both have to be updated.  I used `decoratedPage` instead of `twoColumn` and my build is successful.  They are [line 270](https://github.com/rst2pdf/rst2pdf/blob/main/rst2pdf/pdfbuilder.py#L270) in `if genindex` block and [line 338](https://github.com/rst2pdf/rst2pdf/blob/main/rst2pdf/pdfbuilder.py#L338) where various domain indexes are generated.

The template to use is hard coded in `pdfbuilder.py` so configuration can't change this.  My recommendation is to change those two locations to use `decoratedPage` instead, while considering adding config options or similar so that sphinx builds can work again.
